### PR TITLE
[FLINK-18067][yarn] Change default value of yarnMinAllocationMB from zero to DEFAULT_RM_S…

### PR DESCRIPTION
**Pull request by Jiayi-Liao**
# What is the purpose of the change

As discussed in https://github.com/apache/flink/pull/11176#discussion_r433629963, the default value of yarnMinAllocationMB is set to zero, which is confusing and may cause ArithmeticException in normalization. 

This PR is trying to fix the confusing default value and add pre-check to avoid ArithmeticException(i.e. / by zero). Also we will cover the case when the yarn-default.xml is removed from classpath.

## Brief change log

* Make the default value to YarnConfiguration.DEFAULT_RM_SCHEDULER_MINIMUM_ALLOCATION_MB, not 0
* Add pre-check for yarnMinAllocationMB, it should be greater than 0.

## Verifying this change

This should be covered in existing unit tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: yes
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Comment by Jiayi-Liao** [comment]
@tillrohrmann  @wangyang0918 
Could you help review this change? Thanks for your time.

File: flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java [hunk 1]
```diff
@@ -508,7 +508,9 @@ public void killCluster(ApplicationId applicationId) throws FlinkException {
 			throw new YarnDeploymentException("Could not retrieve information about free cluster resources.", e);
 		}
 
-		final int yarnMinAllocationMB = yarnConfiguration.getInt(YarnConfiguration.RM_SCHEDULER_MINIMUM_ALLOCATION_MB, 0);
+		final int yarnMinAllocationMB = yarnConfiguration.getInt(
```
[the remark this answers is not in the document]
Comment (Jiayi-Liao) [reply flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java:511]: I put in `validateClusterResources` because I thought it's a part of the validation, but I'm also okay with your way.

**Comment by Jiayi-Liao** [comment]
@wangyang0918  Thanks for your review. I've addressed your comments and also update the description. Please take another look when you can spare time.

File: flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java [hunk 2]
```diff
@@ -561,7 +565,6 @@ private ClusterSpecification validateClusterResources(
 		int yarnMinAllocationMB,
 		Resource maximumResourceCapability,
 		ClusterResourceDescription freeClusterResources) throws YarnDeploymentException {
-
```
Comment (wangyang0918) [inline flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java:564]: nit: The line change is not related.

Comment (Jiayi-Liao) [reply flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java]: done.

**Comment by wangyang0918** [comment]
The changes look good to me. cc @TisonKun @tillrohrmann could you please have a final check and help with merging?

File: flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java [hunk 3]
```diff
@@ -508,7 +508,11 @@ public void killCluster(ApplicationId applicationId) throws FlinkException {
 			throw new YarnDeploymentException("Could not retrieve information about free cluster resources.", e);
 		}
 
-		final int yarnMinAllocationMB = yarnConfiguration.getInt(YarnConfiguration.RM_SCHEDULER_MINIMUM_ALLOCATION_MB, 0);
+		final int yarnMinAllocationMB = yarnConfiguration.getInt(
+				YarnConfiguration.RM_SCHEDULER_MINIMUM_ALLOCATION_MB,
+				YarnConfiguration.DEFAULT_RM_SCHEDULER_MINIMUM_ALLOCATION_MB);
+		Preconditions.checkArgument(yarnMinAllocationMB > 0, "The minimum allocation memory "
+				+ "(" + yarnMinAllocationMB + " MB) configured via 'yarn.scheduler.minimum-allocation-mb' should be greater than 0.");
```
Comment (xintongsong) [inline flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java:515]: The string literal can be replaced with `YarnConfiguration.RM_SCHEDULER_MINIMUM_ALLOCATION_MB`.

File: flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (xintongsong) [reply flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java:511]: I think it would be good to always try to throw an exception with the most descriptive type if possible. IMO, this should have higher priority than keeping the codes concise.

In this case, I would suggest to throw either a `YarnDeploymentException` or a `IllegalConfigurationException`, rather than `IllegalStateException` that `Preconditions.checkArgument` throws.

**Comment by Jiayi-Liao** [comment]
@xintongsong Thanks for the review. I've addressed your comments and the PR should be okay now.

**Comment by xintongsong** [comment]
@flinkbot run azur

**Comment by xintongsong** [comment]
@flinkbot run azure
